pandas/io/json/_json.py

```diff
@@ -726,7 +742,23 @@ def read(self):
             obj = concat(self)
         elif self.lines:
             data = ensure_str(self.data)
-            obj = self._get_object_parser(self._combine_lines(data.split("\n")))
+            if self.nrows:
+                compiled_pattern = re.compile("\n")
```

Comment:
I think this is a little more complicated than it needs to be. If you use a pattern that grabs the text preceding the newline line `r".*\n"` I think can then just call `x.group(0)` during iteration to get what you need, rather than trying to manually slice the string based off of start / end positions

See https://docs.python.org/3/library/re.html#finding-all-adverbs-and-their-positions

Reply:
Thanks a lot. Will make the changes